### PR TITLE
HPCC-31569 Thor CostExecute calc will be incorrect if workersPerPod > 1

### PR DESCRIPTION
**Pull request by shamser**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

### CODE REVIEW

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -13937,13 +13937,16 @@ extern WORKUNIT_API double getThorManagerRate()
 
 extern WORKUNIT_API double getThorWorkerRate()
 {
+    // Note: (bare-metal) the use of getAffinityCpus to get the number of CPUs used by workers
+    // doesn't really make sense since the caller is likely to be running on thor manager (so it will
+    // return cpu affinity for the manager, rather than for the worker).  This needs rethinking.
```
Comment (jakesmith) [inline common/workunit/workunit.cpp:13942]: agreed. Is there a JIRA to track this? (can you link it to HPCC-31569)

**Review by jakesmith — COMMENTED** [review]
I think the change is correct, but it does beg the question, why was it changed in https://hpccsystems.atlassian.net/browse/HPCC-28296 ?
Unfortunately, that JIRA doesn't describe the problem sufficiently to tell what the motivation was.

**Comment by jakesmith** [comment]
@shamser - look correct to me, but could do with a fuller description of the problem in title+description + a associating a JIRA for the bare-metal issue.

File: common/workunit/workunit.cpp (hunk 1 above)
Comment (shamser) [reply common/workunit/workunit.cpp:13942]: done. I've created https://hpccsystems.atlassian.net/browse/HPCC-31572 and linked to this issue.

**Review by jakesmith — APPROVED** [review]
@shamser - looks correct.

**Pull request merged**